[DDW-648] Move funds from legacy wallet CTA

yakovkaravelov opened the pull request:
This PR implements legacy wallet notification and badge to enable moving funds from legacy wallet CTA.
[YouTrack Ticket](https://iohk.myjetbrains.com/youtrack/issue/DDW-648)

## Todos
- [x] Legacy badge in the sidebar
- [x] Legacy badge in the wallet header
- [x] Red notification with two buttons
- [x] Inject Japanese translations

## Screenshots
![image](https://user-images.githubusercontent.com/6207747/58946884-80a6d100-878f-11e9-81f4-0941f1e069ee.png)

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1559727509002900)
- [x] Test new legacy badge and notification UI components on storybook by visiting `WalletScreens` => `Summary` story along with `Legacy` sub menu item in sidebar.
- [x] Test all other UI components if they are not broken.
---

## Review Checklist

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

nikolaglumac commented on source/renderer/app/components/layout/TopBar.scss:
```diff
@@ -20,6 +20,10 @@
+      position: relative;
+      display: flex;
+      flex-direction: column;
+      align-items: center;
```
@yakovkaravelov please order CSS rules alphabetically 🙏

yakovkaravelov replied:
Fixed all the css rules sorting.

nikolaglumac commented on source/renderer/app/components/layout/TopBar.scss:
```diff
@@ -59,5 +70,8 @@
     margin-right: 24px;
+    display: flex;
+    justify-content: center;
+    align-items: center;
```
@yakovkaravelov please order CSS rules alphabetically 🙏

nikolaglumac commented on source/renderer/app/components/layout/TopBar.js:
```diff
@@ -26,13 +27,18 @@ export default class TopBar extends Component<Props> {
-      <div className={styles.walletInfo}>
-        <div className={styles.walletName}>{activeWallet.name}</div>
-        <div className={styles.walletAmount}>
+      <span className={styles.walletInfo}>
```
@yakovkaravelov why did we need to change `<div>`s to `<span>`s?

nikolaglumac replied:
@yakovkaravelov I still didn't get your answer for this question...

yakovkaravelov replied:
@nikolaglumac 
I changed them to span to make the width of the elements shrink to content size
Needed to position badge to right edge of text content

nikolaglumac commented on source/renderer/app/components/layout/TopBarLayout.scss:
```diff
@@ -19,5 +19,5 @@ $topBarHeight: 84px;
 
 .content {
-  height: calc(100% - #{$topBarHeight});
+  height: auto;
```
@yakovkaravelov are you sure this is right? Please check how the screens which have little content look like. I am afraid the `.content` wrapper will not extend to the available height in this case.

nikolaglumac commented on source/renderer/app/components/notifications/LegacyBadge.scss:
```diff
@@ -0,0 +1,7 @@
+  color: var(--theme-legacy-notification-title-color);
+  border-radius: 2px;
+  padding: 3px 6px;
+  font-size: 10px;
```
@yakovkaravelov please order CSS rules alphabetically 🙏

nikolaglumac commented on source/renderer/app/components/notifications/LegacyNotification.scss:
```diff
@@ -0,0 +1,85 @@
+  text-align: center;
+  margin-left: 10px;
+  max-width: 300px;
+  cursor: pointer;
```
@yakovkaravelov please order CSS rules alphabetically 🙏

nikolaglumac commented on source/renderer/app/components/notifications/LegacyNotification.scss:
```diff
@@ -0,0 +1,85 @@
+  background-color: var(--theme-legacy-notification-background-color);
+  box-shadow: 0 5px 20px 0 rgba(0, 0, 0, 0.25);
+  padding: 20px 40px;
+  display: flex;
```
@yakovkaravelov please order CSS rules alphabetically 🙏

nikolaglumac commented on source/renderer/app/components/layout/TopBar.js:
```diff
@@ -26,13 +27,18 @@ export default class TopBar extends Component<Props> {
-      </div>
+        </span>
+        {activeWallet.isLegacy && (
+          <div className={styles.legacyBadge}>
```
@yakovkaravelov why do we need a `<div className={styles.legacyBadge}>` wrapper here? Can't we move all the needed HTML elements into the `<LegacyBadge />` itself?

yakovkaravelov replied:
@nikolaglumac 
This is because we need to position legacy badge in different ways in places.
For example positioning it in sidebar menu and topbar title sections are different and that's why they use wrapper element to style accordingly.

daniloprates replied:
@yakovkaravelov it's better to keep the logic inside the component. It could receive some props for positioning. Maybe only left and right would be enough. Otherwise, an object with the style needed.

nikolaglumac commented on source/renderer/app/components/sidebar/wallets/SidebarWalletMenuItem.js:
```diff
@@ -40,6 +43,11 @@ export default class SidebarWalletMenuItem extends Component<Props> {
           <span className={styles.info}>{info}</span>
           {isRestoreActive ? <ProgressBar progress={restoreProgress} /> : null}
+          {isLegacy && (
+            <div className={styles.legacyBadge}>
```
@yakovkaravelov why do we need a `<div className={styles.legacyBadge}>` wrapper here? Can't we move all the needed HTML elements into the `<LegacyBadge />` itself?

nikolaglumac commented on source/renderer/app/components/sidebar/wallets/SidebarWalletMenuItem.scss:
```diff
@@ -33,6 +33,7 @@
   position: relative;
   text-align: left;
   width: 100%;
+  position: relative;
```
@yakovkaravelov please order CSS rules alphabetically 🙏

nikolaglumac commented on source/renderer/app/components/sidebar/wallets/SidebarWalletMenuItem.scss:
```diff
@@ -63,3 +64,9 @@
+
+.legacyBadge {
+  position: absolute;
+  top: 0;
```
@yakovkaravelov please order CSS rules alphabetically 🙏

nikolaglumac commented on source/renderer/app/domains/Wallet.js:
```diff
@@ -47,6 +47,7 @@ export type WalletProps = {
   hasPassword: boolean,
   passwordUpdateDate: ?Date,
   syncState?: WalletSyncState,
+  isLegacy?: boolean,
```
@yakovkaravelov isLegacy should be mandatory param: 
```
isLegacy: boolean,
```

nikolaglumac commented on source/renderer/app/domains/Wallet.js:
```diff
@@ -57,6 +58,7 @@ export default class Wallet {
   @observable hasPassword: boolean;
   @observable passwordUpdateDate: ?Date;
   @observable syncState: ?WalletSyncState;
+  @observable isLegacy: ?boolean;
```
@yakovkaravelov `isLegacy` should never be null:
```
@observable isLegacy: boolean;
```

nikolaglumac commented on source/renderer/app/types/sidebarTypes.js:
```diff
@@ -6,4 +6,5 @@ export type SidebarWalletType = {
   isConnected: boolean,
   isRestoreActive?: boolean,
   restoreProgress?: number,
+  isLegacy?: ?boolean,
```
@yakovkaravelov please use the following type declaration:
```
isLegacy: boolean,
```

yakovkaravelov replied:
Fixed.

nikolaglumac commented on storybook/stories/support/StoryLayout.js:
```diff
@@ -153,31 +158,42 @@ export default class StoryLayout extends Component<Props> {
+      </TopBar>
+    );
+
+    return activeWallet.isLegacy ? (
```
@yakovkaravelov I don't like this approach. Can't we make the legacy notification to be a part of the TopBar component? That should be much cleaner...

yakovkaravelov replied:
Fixed this.

yakovkaravelov commented:
> @yakovkaravelov I have added a couple of change requests.
> 
> Also, I think we have agreed to pass fake `isLegacy: false` here: https://github.com/input-output-hk/daedalus/blob/feature/ddw-648-ui-move-funds-from-legacy-wallet-cta/source/renderer/app/api/api.js#L1072

Fixed this.

daniloprates commented on storybook/stories/support/StoryProvider.js:
```diff
@@ -20,6 +20,7 @@ export const WALLETS = [
     assurance: WalletAssuranceModeOptions.NORMAL,
     hasPassword: false,
     passwordUpdateDate: new Date(),
+    isLegacy: true,
```
Please, create a third wallet, named Legacy wallet or use the knobs, so we can toggle `isLegacy`

daniloprates commented:
@yakovkaravelov please, fix e2e test `features/select-language.feature`.

nikolaglumac commented on source/renderer/app/components/sidebar/wallets/SidebarWalletMenuItem.js:
```diff
@@ -40,6 +43,15 @@ export default class SidebarWalletMenuItem extends Component<Props> {
           {isRestoreActive ? <ProgressBar progress={restoreProgress} /> : null}
+          {isLegacy && (
+            <LegacyBadge
+              style={{
```
@yakovkaravelov why do we have a style prop here? Can't we sort out styling in the component itself?

yakovkaravelov commented:
@nikolaglumac @daniloprates Please review this.
Finished.
Thanks.

a-rukin commented:
<img width="1331" alt="image" src="https://user-images.githubusercontent.com/2539345/58992891-aadeaa00-87f4-11e9-868a-bba887eebfe1.png">
"This is a legacy wallet which uses legacy addresses..." is regular now, but Light in design 
@nikolaglumac do we use regular for all notification texts?

yakovkaravelov commented:
> <img alt="image" width="1331" src="https://user-images.githubusercontent.com/2539345/58992891-aadeaa00-87f4-11e9-868a-bba887eebfe1.png">
> 
> "This is a legacy wallet which uses legacy addresses..." is regular now, but Light in design @nikolaglumac do we use regular for all notification texts?

@a-rukin Just fixed the font weight values also.
Please review again once builds are completed.

a-rukin commented on source/renderer/app/components/notifications/LegacyNotification.scss:
```diff
@@ -74,6 +76,7 @@
   color: var(--theme-legacy-notification-description-color);
   font-family: var(--font-regular);
```
shouldn't this be "--font-light"?

nikolaglumac commented on source/renderer/app/components/notifications/LegacyBadge.scss:
```diff
@@ -10,7 +10,9 @@
   right: -58.5px;
-  top: 0;
+  top: -2px;
+  border-top-right-radius: none;
```
@yakovkaravelov please sort these CSS rules alphabetically 🙏 
Also, these CSS rules are invalid :(
![Screenshot 2019-06-07 at 09 45 45](https://user-images.githubusercontent.com/376611/59088880-0d28cf00-8909-11e9-83e3-b1b3957b9839.png)
